RAM-75

Paulo-Araujo-Jr opened the pull request:
## feat: episode synchronization improvements ##

### Description:  ###

Updated `EpisodeDto.id` from _Integer_ to _Long_ for better compatibility. Fixed `EpisodeDto.air_date` parsing issue by implementing correct date format ("MMMM d, yyyy", e.g., "October 4, 2015")
:shipit: :+1: :rocket:

Paulo-Araujo-Jr committed:
feat: episode synchronization implementation

MrMesquita commented on src/main/java/com/rickmorty/Services/SyncApiService.java:
```diff
@@ -54,9 +53,43 @@ public void scheduledSync() {
         System.out.println("Sincronização automática concluída!");
     }
 
+    private void syncEpisode(String url) {
```
move a função para depois da função syncCharacters para manter a ordem com base no que é chamado no construtor da classe

MrMesquita reviewed (changes requested):
Deixei umas sugestões na pr

Paulo-Araujo-Jr committed:
refactor: episode synchronization

1. Reorganized the function order by moving the episode synchronization function after the character synchronization function to maintain the order as per the class constructor.
2. Added spacing between the check for `episodeList.results() == null` to improve readability.
3. Added extraction of character IDs from URLs, similar to how location IDs are extracted, and queried the database to fetch the characters related to the episode.
4. Established the relationship between characters and episodes by using `setCharacters` to correctly associate the characters with the episode.

MrMesquita commented on src/main/java/com/rickmorty/Services/SyncApiService.java:
```diff
@@ -138,7 +146,67 @@ private void syncCharacters(String url) {
+            try {
+                EpisodeModel episodeModel = this.episodeService.saveEpisodeByDto(episode);
+                if (episodeModel != null) {
+                    System.out.println("Episódio: " + episodeModel.getName() + " salvo!");
```
move o print pro final do if, pq se der algum erro dps desse print e não salvar as informações vão ficar redundantes

MrMesquita reviewed (changes requested):
só um ajuste mínimo para fazer, de resto ta perfeito

brunoliratm reviewed (approved):
A sincronização dos dados da API externa para o banco está funcionando perfeitamente.
